libbeat/common/event.go

```diff
@@ -158,6 +158,9 @@ func normalizeValue(value interface{}, keys ...string) (interface{}, []error) {
 
 	switch value.(type) {
 	case encoding.TextMarshaler:
+		if reflect.ValueOf(value).Kind() == reflect.Ptr && reflect.ValueOf(value).IsNil() {
```

Comment:
Does it make sense to do this check up on line 135 where we check for plain old `nil`?

Reply:
Right, but should we do this check immediately upon entering `normalizeValue` so that we drop all interfaces storing nil pointers?

Reply:
@andrewkroh: I didn't want to add a run-time check for all fields in the event map.

The 2 cases where the `interface{}` can store a pointer to a nil value are `encoding.TextMarshaler` and `followPointer`. And `followPointer`  already checks for nil pointer
 

If you prefer, I can move the test to the beginning of the function `normalizeValue`.
It could be safer if a new encoder case is added later.

Reply:
> I didn't want to add a run-time check for all fields in the event map.

I suspected that might be a reason for not doing it at method entry.